polymode-compat.el

```diff
@@ -286,15 +286,18 @@ changes."
 
 (defun polymode-fix-desktop-buffer-info (fn buffer)
   "Save polymode buffers without mode prefix."
-  (let ((out (funcall fn buffer))
-        (base (buffer-base-buffer)))
+  (let* ((out (funcall fn buffer))
+         (base (buffer-base-buffer))
+         (base-name (and base
+                         (string-trim-left (buffer-name base)))))
     (with-current-buffer buffer
-      (if (not (and polymode-mode base))
-          out
+      (when (and polymode-mode base)
         (when (car out)
-          (setf (car out) (buffer-name base)))
-        (setf (nth 2 out) (buffer-name base))
-        out))))
+          (setf (car out) base-name))
+        (setf (nth 2 out) base-name))
+      (when polymode-mode
+        (setf (nth 3 out) (intern (substring (oref pm/polymode :name) 0 -1))))
```

Comment:
:name is an unique identifier of the polymod object, not the mode. So this is surely not correct. There is `-minor-mode` slot which holds the symbol of the mode which should be installed when polymode is installed.

Reply:
Thanks, I didn't notice this slot.